diff --git a/dist/esm/index.js b/dist/esm/index.js
index 25cf5bc1aff674ecbc5b2a032fd036a91699870c..9de4e824f403891d26fa63d7b32faa956b7b372c 100644
--- a/dist/esm/index.js
+++ b/dist/esm/index.js
@@ -410,6 +410,10 @@ class Virtualizer {
         );
         this.unsubs.push(
           this.options.observeElementOffset(this, (offset, isScrolling) => {
+            if (isScrolling && this._intendedScrollOffset === null && offset === this.scrollOffset) {
+              return
+            }
+
             if (this._intendedScrollOffset !== null && Math.abs(offset - this._intendedScrollOffset) < 1.5) {
               offset = this._intendedScrollOffset;
             }
