diff --git a/dist/es6/popup/lib/hooks/usePopupStack.js b/dist/es6/popup/lib/hooks/usePopupStack.js
index 60d0bd70cc2c1f23c10d695a8c4a6fb8f94ee270..aa2bc84de38be06edfa1e04cc0e46f96140fefdd 100644
--- a/dist/es6/popup/lib/hooks/usePopupStack.js
+++ b/dist/es6/popup/lib/hooks/usePopupStack.js
@@ -59,7 +59,7 @@ export const usePopupStack = (ref, target) => {
     // Without this, the timing is unpredictable when mixed with other frameworks. Other frameworks
     // should also register as soon as the element is available
     React.useLayoutEffect(() => {
-        if (popupRef !== localRef.current) {
+        if (!localRef.current) {
             throw Error(`The 'ref' passed to usePopupStack should not be applied to a React element. This will cause a runtime error where the PopupStack and React compete for the element. Instead use ReactDOM.createPortal(<YourComponent />, ref.current)`);
         }
         const targetEl = target
diff --git a/popup/lib/hooks/usePopupStack.ts b/popup/lib/hooks/usePopupStack.ts
index 788baa34d3a1f06fd37d29701ac0331ebc3f888b..7baec7a4fd2e0de7570eda0f7c6eb9bedac26575 100644
--- a/popup/lib/hooks/usePopupStack.ts
+++ b/popup/lib/hooks/usePopupStack.ts
@@ -65,7 +65,7 @@ export const usePopupStack = <E extends HTMLElement>(
   // Without this, the timing is unpredictable when mixed with other frameworks. Other frameworks
   // should also register as soon as the element is available
   React.useLayoutEffect(() => {
-    if (popupRef !== localRef.current) {
+    if (!localRef.current) {
       throw Error(
         `The 'ref' passed to usePopupStack should not be applied to a React element. This will cause a runtime error where the PopupStack and React compete for the element. Instead use ReactDOM.createPortal(<YourComponent />, ref.current)`
       );
