diff --git a/dist/es6/popup/lib/hooks/usePopupStack.js b/dist/es6/popup/lib/hooks/usePopupStack.js
index 724a3f949ecdca32b57f58976a9bdc416794a992..2afe73c1b151a63ab9533cf0ae4cd1d96bd88403 100644
--- a/dist/es6/popup/lib/hooks/usePopupStack.js
+++ b/dist/es6/popup/lib/hooks/usePopupStack.js
@@ -63,7 +63,7 @@ export const usePopupStack = (ref, target) => {
     // Without this, the timing is unpredictable when mixed with other frameworks. Other frameworks
     // should also register as soon as the element is available
     React.useLayoutEffect(() => {
-        if (popupRef !== localRef.current) {
+        if (!localRef.current) {
             throw Error(`The 'ref' passed to usePopupStack should not be applied to a React element. This will cause a runtime error where the PopupStack and React compete for the element. Instead use ReactDOM.createPortal(<YourComponent />, ref.current)`);
         }
         const targetEl = target
diff --git a/popup/lib/hooks/usePopupStack.ts b/popup/lib/hooks/usePopupStack.ts
index b688c4bdc73962e23ae8dffed336672c63c4f56b..fdaa8cdf4230531686dd044b510e2d45165f765b 100644
--- a/popup/lib/hooks/usePopupStack.ts
+++ b/popup/lib/hooks/usePopupStack.ts
@@ -69,7 +69,7 @@ export const usePopupStack = <E extends HTMLElement>(
   // Without this, the timing is unpredictable when mixed with other frameworks. Other frameworks
   // should also register as soon as the element is available
   React.useLayoutEffect(() => {
-    if (popupRef !== localRef.current) {
+    if (!localRef.current) {
       throw Error(
         `The 'ref' passed to usePopupStack should not be applied to a React element. This will cause a runtime error where the PopupStack and React compete for the element. Instead use ReactDOM.createPortal(<YourComponent />, ref.current)`
       );
