diff --git a/dist/es6/popup/lib/hooks/usePopupStack.js b/dist/es6/popup/lib/hooks/usePopupStack.js
index 2f20cdc49bc1be1c00f86ac5bc73856e81efdb5f..2ba3e6fefc75a757b9020a78abf6793afc007c5e 100644
--- a/dist/es6/popup/lib/hooks/usePopupStack.js
+++ b/dist/es6/popup/lib/hooks/usePopupStack.js
@@ -56,7 +56,7 @@ export const usePopupStack = (ref, target) => {
     // Without this, the timing is unpredictable when mixed with other frameworks. Other frameworks
     // should also register as soon as the element is available
     React.useLayoutEffect(() => {
-        if (popupRef !== localRef.current) {
+        if (!localRef.current) {
             throw Error(`The 'ref' passed to usePopupStack should not be applied to a React element. This will cause a runtime error where the PopupStack and React compete for the element. Instead use ReactDOM.createPortal(<YourComponent />, ref.current)`);
         }
         const targetEl = target
diff --git a/popup/lib/hooks/usePopupStack.ts b/popup/lib/hooks/usePopupStack.ts
index cd083e4dd3ad47c87ec92b915a11f1129cbb0a0f..809fbc7be9fa09b8d6e2e299d72cb7108f2b6b81 100644
--- a/popup/lib/hooks/usePopupStack.ts
+++ b/popup/lib/hooks/usePopupStack.ts
@@ -62,7 +62,7 @@ export const usePopupStack = <E extends HTMLElement>(
   // Without this, the timing is unpredictable when mixed with other frameworks. Other frameworks
   // should also register as soon as the element is available
   React.useLayoutEffect(() => {
-    if (popupRef !== localRef.current) {
+    if (!localRef.current) {
       throw Error(
         `The 'ref' passed to usePopupStack should not be applied to a React element. This will cause a runtime error where the PopupStack and React compete for the element. Instead use ReactDOM.createPortal(<YourComponent />, ref.current)`
       );
